mesonbuild/mlog.py

```diff
@@ -50,9 +50,12 @@ def initialize(logdir):
 def shutdown():
     global log_file
     if log_file is not None:
+        path = log_file.name
         exception_around_goer = log_file
         log_file = None
         exception_around_goer.close()
+        return path
+    return None
```

Comment:
Why?

Reply:
Because commit "Use a single ArgumentParser for all subcommands" uses it.

It is nicer to get the log file from mlog.shutdown() instead of doing `logfile = os.path.join(app.build_dir, environment.Environment.log_dir, mlog.log_fname)` because it's mlog who decide the path where it store its logs, not mesonmain.